db/types/alteration.py

```diff
@@ -25,15 +27,20 @@ def get_supported_alter_column_types(engine):
 
 
 def alter_column_type(
-        schema, table_name, column_name, target_type_str, engine
+        schema, table_name, column_name, target_type_str, engine, conn=None
 ):
     _preparer = engine.dialect.identifier_preparer
     supported_types = get_supported_alter_column_types(engine)
     target_type = supported_types.get(target_type_str.lower())
-    with engine.begin() as conn:
+
+    with ExitStack() as stack:
+        if conn is not None:
+            stack.enter_context(conn.begin())
+        else:
+            conn = stack.enter_context(engine.begin())
```

Comment:
For my own information:  Why is this move necessary?

Reply:
To access the temp table, we have to start our transaction using the connection that made the temp table. So here we start the transaction using a connection if there is a connection, and the engine otherwise. (Hopefully will be getting rid of this with the move from temp tables though.)